client/my-sites/upgrades/checkout-thank-you/failed-purchase.jsx

```diff
@@ -0,0 +1,71 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import Gridicon from 'components/gridicon';
+import { localize } from 'i18n-calypso';
+import PurchaseDetail from 'components/purchase-detail';
+
+const FailedPurchase = ( { failedPurchases, translate } ) => {
+	function getDescription() {
```

Comment:
Yeah, in this form it doesn't make more sense than a regular variable. I was thinking of putting it outside this stateless component as a normal function (probably accepting `failedPurchases` and `translate`). I agree it's not exactly elegant, but IMHO this component is just to large without it. Why didn't you just extend and use `CheckoutThankYouHeader` here? It seems a lot of the code is just copied from it.